src/ess/reduce/ui.py

```diff
@@ -232,3 +234,136 @@ def refresh_workflow_box(change) -> None:
+    new_parameter_values: dict[str | type, dict[str, Any]],
+) -> None:
+    """Set the values of the input widgets in the target widget.
+
+    Example
+    -------
+    {
+        'WavelengthBins': {
+            'fields': {'start': 1.0, 'stop': 14.0, 'nbins': 500}
+        }
+    }
+
+    Parameters
+    ----------
+    widget:
+        The widget containing the input widgets.
+    new_parameter_values:
+        A dictionary of parameter values to set for the input widgets.
+        The keys are the parameter names or the type(Key in Sciline graphs).
+        The values are dictionaries with the following keys
+
+        - 'fields':
+            A dictionary of field names and values to set.
+            i.e. {'dim': 'tof', 'unit': 'mm'}
+            The field values will be set if the field exists in the widget.
+            Otherwise, the field value will be ignored and a warning will be raised.
+        - 'value':
+            If the widget has a value setter, the value will be set directly.
+            Otherwise, it will be ignored and a warning will be raised.
+
+    Raises
+    ------
+    TypeError:
+        If the target_widget is not a WorkflowWidget or a ParameterBox.
+
+    """
+    parameter_box = _get_parameter_box(widget)
+    # Walk through the existing input widgets and set the values
+    for node, widget in parameter_box._input_widgets.items():
+        new_internal_values = new_parameter_values.get(
+            node, new_parameter_values.get(parameter_box._input_registry[node].name, {})
+        )
+        # Parse the new values and corresponding fields in the existing widget
+        new_fields: dict[str, Any] = new_internal_values.get('fields', {})
+        new_fields_keys = set(new_fields.keys())
+        widget_fields = getattr(widget, "fields", {})
+        # Extract valid fields
+        valid_fields = new_fields_keys & set(widget_fields.keys())
+        # Warn for invalid fields
+        invalid_fields = new_fields_keys - valid_fields
+        for field_name in invalid_fields:
+            warning_msg = f"Cannot set field '{field_name}' for parameter '{node}'."
+            " The field does not exist in the widget. "
+            "The field value will be ignored."
+            warnings.warn(warning_msg, UserWarning, stacklevel=1)
+        # Set the valid fields
+        for field_name in valid_fields:
+            widget_fields[field_name].value = new_fields[field_name]
+
+        # Set the high-level value if needed and possible.
+        # It is set after the fields to prioritize the high-level value.
+        if (self_value := new_internal_values.get('value')) is not None:
+            if _has_widget_value_setter(widget):
+                widget.value = self_value
+            else:
+                warning_msg = f"Cannot set value for parameter '{node}'. "
+                "The widget does not have a value setter. The value will be ignored."
+                warnings.warn(warning_msg, UserWarning, stacklevel=1)
```

Comment:
Then we require all widgets to have `set_value` method.
It might also make it easier to handle optional/switchable wrapper... hmm... I'll try that though.